python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -967,10 +970,16 @@ def _setFlushed(self):
         """Un-Set the dirty flag all of the way down the schema."""
         if self._schema:
             for k in self._schema.allItemNames():
+                ## Avoid attributes the likes of job.master which crawl back up the tree
+                if self._schema[k].hasProperty('visitable'):
+                    if (not self._schema[k].getProperties()['visitable'] or self._schema[k].getProperties()['visitable'] == 0):
+                        continue
+                if self._schema[k].hasProperty('transient'):
+                    if(self._schema[k].getProperties()['transient'] or self._schema[k].getProperties()['transient'] == 1):
```

Comment:
I agree that they shouldn't be defined as `0` and `1` but `1` and `True` are almost entirely compatible (likewise for `0` and `False`). We should write code which works with any truthy or falsey values so that when we replace all the integers with booleans it will keep working.

The `or foo == 0` check above is entirely unnecessary as if `foo` is ever `0` then it will also evaluate to `False` in a boolean context.